File: consumer/src/test/java/no/nav/sbl/dialogarena/soknadinnsending/consumer/unleash/ByInstanceIdStrategyTest.java
package no.nav.sbl.dialogarena.soknadinnsending.consumer.unleash;

import org.junit.After;
import org.junit.Test;

import java.util.Map;

import static org.junit.Assert.*;

public class ByInstanceIdStrategyTest {

    private final ByInstanceIdStrategy strategy = new ByInstanceIdStrategy();

    @After
    public void tearDown() {
        System.clearProperty("unleash_instance_id");
    }

    @Test
    public void shouldReturnFalse_instanceIdNotInMap() {
        System.setProperty("unleash_instance_id", "local");
        var parameters = Map.of("instance.id", "dev-sbs,dev-sbs-intern");

        assertFalse(strategy.isEnabled(parameters));
    }

    @Test
    public void shoudReturnTrue_instanceIdInMap() {
        System.setProperty("unleash_instance_id", "dev-sbs");
        var parameters = Map.of("instance.id", "dev-sbs,dev-sbs-intern");

        assertTrue(strategy.isEnabled(parameters));
    }

}

File: src/test/kotlin/no/nav/sosialhjelp/soknad/app/featuretoggle/unleash/ByInstanceIdStrategyTest.kt
package no.nav.sosialhjelp.soknad.app.featuretoggle.unleash

import io.getunleash.UnleashContext
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class ByInstanceIdStrategyTest {
    private val unleashContext: UnleashContext = UnleashContext.builder().build()

    @Test
    fun shouldReturnFalse_instanceIdNotInMap() {
        val byInstanceIdStrategy = ByInstanceIdStrategy("local")
        val parameters = mutableMapOf("instance.id" to "dev,mock")
        assertThat(byInstanceIdStrategy.isEnabled(parameters, unleashContext)).isFalse
    }

    @Test
    fun shoudReturnTrue_instanceIdInMap() {
        val byInstanceIdStrategy = ByInstanceIdStrategy("dev")
        val parameters = mutableMapOf("instance.id" to "dev,mock")
        assertThat(byInstanceIdStrategy.isEnabled(parameters, unleashContext)).isTrue
    }
}
